### DISCUSSION TITLE
Video recording feature

### DISCUSSION THREAD

**Discussion by binarykitchen** [WordPress/gutenberg#51211] [Ideas]
Hello again,

Wondering if you have a package to video record and insert a video right away in the browser without any detours?

**Comment by binarykitchen** [comment]
@wojtekLs Isn't that too painful, feels like a detour? And what about privacy, you don't want to host videos on YT solely for posts on another seerver?

**Comment by binarykitchen** [comment]
@wojtekLs understand. Might work fine like that.

It is just this:
- What if you want to host and server all the data (including videos) from the same server?
- YT processes all videos for internal use, recommendations, and they are never 100% private nor secure.
- Not everyone knows how to use YT's sharing and embedding features properly.
- Accessibility: Deaf people intending to post in Sign Language. It's an obstacle.
- Usability: Pretend I want to say something over video quickly, just 10 seconds, while I'm busy. A detour over YT? No, thanks.

**Reply by binarykitchen** [reply]
@wojtekLs How can screen readers and grabbers assist with Sign Language?

**Reply by binarykitchen** [reply]
I can't think of any screen readers being able to interpret Sign Language into spoken language.

**Comment by devfle** [comment]
Hey @binarykitchen,

there is currently no option to directly create a video in the Gutenberg editor. You have to upload it at least to your media library in WordPress. But you can host it on your own instance. To embed the Video you can use the "Video Block" or use the "Custom HTML" Block. You can even create your own [Block](https://developer.wordpress.org/block-editor/how-to-guides/block-tutorial/writing-your-first-block-type/) or create a Preset, called ["Block Pattern"](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-patterns/#block-patterns) of your block configuration.

**Reply by binarykitchen** [reply]
@devfle thanks for the pointer, that looks better. But still, not being able to directly record a video is still a disadvantage.

**Comment by binarykitchen** [comment]
Wondering if you would welcome a new Gutenberg plugin which enables direct video recording, on the same service instance? 

I could help with that.

**Reply by binarykitchen** [reply]
"moreless"? - That's a new word for me.

"I can code it?"  - Yep, there is already a working solution out there :)
